metrics/config.go

```diff
@@ -193,19 +199,34 @@ func createMetricsConfig(ops ExporterOptions, logger *zap.SugaredLogger) (*metri
 		scc := newStackdriverClientConfigFromMap(m)
 		mc.stackdriverClientConfig = *scc
 		mc.isStackdriverBackend = true
+		var allowCustomMetrics bool
+		var err error
 		mc.stackdriverMetricTypePrefix = path.Join(mc.domain, mc.component)
 
-		mc.stackdriverCustomMetricsSubDomain = defaultCustomMetricSubDomain
-		if sdcmd, ok := m[StackdriverCustomMetricSubDomainKey]; ok && sdcmd != "" {
-			mc.stackdriverCustomMetricsSubDomain = sdcmd
+		customMetricsSubDomain := m[StackdriverCustomMetricSubDomainKey]
+		if customMetricsSubDomain == "" {
+			customMetricsSubDomain = defaultCustomMetricSubDomain
 		}
-		mc.stackdriverCustomMetricTypePrefix = path.Join(customMetricTypePrefix, mc.stackdriverCustomMetricsSubDomain, mc.component)
-		if ascmStr, ok := m[AllowStackdriverCustomMetricsKey]; ok && ascmStr != "" {
-			ascmBool, err := strconv.ParseBool(ascmStr)
+		mc.stackdriverCustomMetricTypePrefix = path.Join(customMetricTypePrefix, customMetricsSubDomain, mc.component)
+		if ascmStr := m[AllowStackdriverCustomMetricsKey]; ascmStr != "" {
+			allowCustomMetrics, err = strconv.ParseBool(ascmStr)
 			if err != nil {
 				return nil, fmt.Errorf("invalid %s value %q", AllowStackdriverCustomMetricsKey, ascmStr)
 			}
-			mc.allowStackdriverCustomMetrics = ascmBool
+		}
+
+		if !allowCustomMetrics {
+			servingOrEventing := metricskey.KnativeRevisionMetrics.Union(
```

Comment:
I optimized this a bit, instead of searching two sets on each Record, we store one larger set, which should be slightly more efficient. (One `log(n)` rather than `2*log(n/2)`.)

If we wanted to be even smarter, we could filter this set by `ms.stackdriverMetricTypePrefix`, but that seems less important than factoring this out of `config.go`.